ui/apps/platform/src/Containers/Dashboard/Widgets/ComplianceLevelsByStandard.cy.jsx

```diff
@@ -0,0 +1,137 @@
+    PCI_DSS_3_2: ['PCI DSS 3.2.1', 4, 6],
+    'ocp4-cis': ['ocp4-cis', 3, 7],
+    'ocp4-cis-node': ['ocp4-cis-node', 2, 8],
+};
+
+const mock = {
+    data: {
+        controls: {
+            results: Object.entries(standards).map(([id, [, numFailing, numPassing]]) => ({
+                aggregationKeys: [{ id, scope: 'STANDARD' }],
+                numFailing,
+                numPassing,
+                numSkipped: 0,
+                unit: 'CONTROL',
+            })),
+        },
+        complianceStandards: Object.entries(standards).map(([id, [name]]) => ({
+            id,
+            name,
+        })),
+    },
+};
+
+beforeEach(() => {
+    localStorage.clear();
+});
+
+const setup = () => {
+    const history = createBrowserHistory();
+
+    cy.intercept('POST', '/api/graphql?opname=getAggregatedResults', (req) => {
+        req.reply(mock);
+    });
+
+    cy.mount(
+        <Router history={history}>
+            <ApolloProvider client={configureApolloClient()}>
+                <ComplianceLevelsByStandard />
+            </ApolloProvider>
+        </Router>
+    );
+};
+
+describe('<ComplianceLevelsByStandard />', () => {
+    it('should render graph bars correctly order by compliance percentage', () => {
+        setup();
+
+        const standardNames = Object.values(standards).map(([name]) => name);
+        const titlesRegex = new RegExp(`${standardNames.join('|')}`);
+
+        // Default is ascending
+        // Note that we use negative indices here because the order in the DOM (bottom->top) is the opposite from
+        // how that chart is displayed to the user (top->bottom)
+        cy.findAllByText(titlesRegex).eq(-1).should('have.text', 'CIS Kubernetes v1.5');
+        cy.findAllByText(titlesRegex).eq(-2).should('have.text', 'HIPAA 164');
+        cy.findAllByText(titlesRegex).eq(-3).should('have.text', 'NIST SP 800-190');
+        cy.findAllByText(titlesRegex).eq(-4).should('have.text', 'NIST SP 800-53');
+        cy.findAllByText(titlesRegex).eq(-5).should('have.text', 'PCI DSS 3.2.1');
+        cy.findAllByText(titlesRegex).eq(-6).should('have.text', 'ocp4-cis');
+
+        // Sort by descending
+        cy.findByLabelText('Options').click();
+        cy.findByText('Descending').click();
+
+        cy.findAllByText(titlesRegex).eq(-1).should('have.text', 'ocp4-cis-node');
+        cy.findAllByText(titlesRegex).eq(-2).should('have.text', 'ocp4-cis');
+        cy.findAllByText(titlesRegex).eq(-3).should('have.text', 'PCI DSS 3.2.1');
+        cy.findAllByText(titlesRegex).eq(-4).should('have.text', 'NIST SP 800-53');
+        cy.findAllByText(titlesRegex).eq(-5).should('have.text', 'NIST SP 800-190');
+        cy.findAllByText(titlesRegex).eq(-6).should('have.text', 'HIPAA 164');
+    });
+
+    it('should visit the correct pages when widget links are clicked', () => {
+        setup();
+
+        cy.findByText('View all').click();
+        cy.url().should('include', complianceBasePath);
```

Comment:
Is it possible to write as `cy.location()` similar to integration tests?

https://docs.cypress.io/api/commands/url

> This is an alias of [cy.location('href')](https://docs.cypress.io/api/commands/location)

Reply:
It looks like the only difference between the two is that `cy.url()` allows passing a parameter of `{ decode: true }` that decodes the URL text. I used that here in order to make the comparison to the expected search parameters a little cleaner.

Reply:
Aha, to deal with `%20` for space in search query string?

Could `cy.log().should('include', …)` be `cy.location('pathname').should('eq', …)` or same problem?

Reply:
Same problem with `cy.location('search')`. It looks like this nicety was only added for `cy.url()` specifically. https://docs.cypress.io/guides/references/changelog#8-4-0